scripts/build-layout-scales.js

```diff
@@ -0,0 +1,114 @@
+       *
+       * @example
+       * <div class='grid grid--gut12'>
+       *   <div class='col col--2'>
+       *     <div class='border border--darken10'>col--2</div>
+       *   </div>
+       *   <div class='col col--4'>
+       *     <div class='border border--darken10'>col--4</div>
+       *   </div>
+       *   <div class='col col--6'>
+       *     <div class='border border--darken10'>col--6</div>
+       *   </div>
+       * </div>
+       * @memberof Grid
+       * @group
+       */`
+    );
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .grid--gut${scale} {
+          margin-left: -${scale}px !important;
+        }
+      `);
+    }, '');
+    css += '/** @endgroup */\n';
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .grid--gut${scale} > .col {
+          padding-left: ${scale}px !important;
+        }
+      `);
+    }, '');
+    css += '\n@media (' + mediaQueries['--m-screen'] + ') {\n';
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .grid--gut${scale}-mm {
+          margin-left: -${scale}px !important;
+        }
+      `);
+    }, '');
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .grid--gut${scale}-mm > .col {
+          padding-left: ${scale}px !important;
+        }
+      `);
+    }, '\n');
+    css += '\n}\n';
+    css += '\n@media (' + mediaQueries['--l-screen'] + ') {\n';
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .grid--gut${scale}-ml {
+          margin-left: -${scale}px !important;
+        }
+      `);
+    }, '');
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .grid--gut${scale}-ml > .col {
+          padding-left: ${scale}px !important;
+        }
+      `);
+    }, '\n');
+    css += '\n}\n';
+    css += '\n@media (' + mediaQueries['--xl-screen'] + ') {\n';
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .grid--gut${scale}-mxl {
+          margin-left: -${scale}px !important;
+        }
+      `);
+    }, '');
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .grid--gut${scale}-mxl > .col {
+          padding-left: ${scale}px !important;
+        }
+      `);
+    }, '\n');
+    css += '\n}\n';
+
+    return css;
+  };
+
+  let result = '\n/* Layout scales */\n';
+
+  Object.keys(variantGenerators).forEach((d) => {
+    const scales = layoutScales[d] || layoutScales.default;
+    if (scales === null || scales === undefined) return;
+    result += variantGenerators[d](scales);
+  });
+
+  return result;
+}
+
+module.exports = buildLayoutScales;
+
+if (require.main === module) {
+  buildLayoutScales().catch((err) => console.error(err.stack));
```

Comment:
`buildLayoutScales` is not returning a promise (right?) so I think this `catch` doesn't make sense. If we have the same situation in `buildColorVariants` I think it's probably a mistake.